spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroReader.java

```diff
@@ -40,35 +41,51 @@ protected void writeAndValidate(Schema schema) throws IOException {
   }
 
   @Override
-  protected void writeAndValidate(Schema writeSchema, Schema expectedSchema) throws IOException {
-    List<Record> expected = RandomData.generateList(writeSchema, 100, 0L);
-
+  protected void writeAndValidate(
```

Comment:
let me publish this separately, this was as part of porting the support of avro with row lineage

Reply:
Separated into https://github.com/apache/iceberg/pull/13496